src/poetry/core/masonry/builders/builder.py

```diff
@@ -206,10 +203,6 @@ def find_files_to_add(self, exclude_build: bool = True) -> Set["BuildIncludeFile
                 if file.suffix == ".pyc":
                     continue
 
-                if file in to_add:
-                    # Skip duplicates
```

Comment:
`to_add` is a set, it couldn't contain duplicates if it tried.  Just need to be happy with `__hash__` and `__eq__` on the `BuildIncludeFile`.